### ISSUE TITLE
Exception 'NSInvalidArgumentException', reason: '-[NSNull length]: unrecognized selector sent to instance

### ISSUE THREAD

**Issue by tobiasmuecksch** [sarriaroman/photoviewer#161]
Hello,

I'm trying to open an image, that is located under an https:// URL.

Screenshot:

![Screenshot](https://cl.ly/80842873c552/download/Bildschirmfoto%202019-03-05%20um%2014.37.53.png)

Dependency versions:
`"@ionic-native/photo-viewer": "^4.20.0"`
`"com-sarriaroman-photoviewer": "^1.2.1"`
`"cordova-ios": "5.0.0"`
`"ionic-angular": "3.9.2"`

Am I doing something wrong?
How can I fix this error?

**Comment by tobiasmuecksch** [comment]
I do have the same error with Ionic v4:

```
   "@ionic/angular": "^4.0.0",
    "@ionic-native/photo-viewer": "^5.2.0",
    "com-sarriaroman-photoviewer": "1.2.1",
    "cordova-ios": "5.0.0",
```

**Comment by tobiasmuecksch** [comment]
```
"cordova-ios": "4.5.5",
```
Same issue

**Comment by tobiasmuecksch** [comment]
Similar error on Android:

```
Unable to start activity ComponentInfo{com.burningparrot.eeapp/com.sarriaroman.PhotoViewer.PhotoActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'boolean org.json.JSONObject.has(java.lang.String)' on a null object reference
````

**Comment by asa47** [comment]
Hello @tobiasmuecksch, I have looked at the code, and it looks like all the parameters in the 'options' variable are required. If you don't add them, it will trigger an exception.

```
const options = {
	share: true, // default is false
	closeButton: true, // default is true
	copyToReference: true, // default is false
        headers: "",  // If it is not provided, it will trigger an exception
        piccasoOptions: { } // If it is not provided, it will trigger an exception
};

```
Please see the following:
https://github.com/sarriaroman/photoviewer/blob/3fe843a9000fa19215530b44c214456acff415b4/www/PhotoViewer.js#L24

**Comment by tobiasmuecksch** [comment]
@asa47 Thank you so much for your response. I will try it in the next few hours and let you know if the issue is resolved.

**Comment by tobiasmuecksch** [comment]
@asa47 That indeed fixed the issue. Thank you so much. I'll create a PR to fix the readme. Until then I'll leave this issue open.

**Comment by halindrome** [comment]
FWIW this is a breaking change.   Patching the documentation is great, but everyone who uses this plugin is suddenly and unexpectedly going to have their app crash.  Really should patch the underlying code to not require all the parameters.

**Comment by mirko77** [comment]
@halindrome I agree. I had my app crash and I was wondering why, here I am.

**Comment by ejerskov** [comment]
@asa47 Thanks alot

**Comment by aacassandra** [comment]
Hey, it's 2021 now, and the problem still persists. but can still be overcome with complete parameter options. btw thankyou @asa47

**Comment by mirko77** [comment]
@aacassandra it's 2021, time to switch to Flutter

**Comment by aacassandra** [comment]
@mirko77 , Yes, you are right 🎉   . but since Vuejs released version 3, Ionic released a stable version of support for @ ionic/vue. We know that vue is a mix between angularjs and reactjs, which the developers are starting to be interested in. And plus the release of the native runtime Capasitor v2, which previously used Cordova. Also as the years have gone on, the devices have become more sophisticated. Whenever I build a plugin or other tool, it's still useful to speed up other projects (frontend, backend & other nodejs), unlike the standalone flutter with Dart. I bet more on ionic development.

**Comment by mirko77** [comment]
> @mirko77 , Yes, you are right 🎉 . but since Vuejs released version 3, Ionic released a stable version of support for @ ionic/vue. We know that vue is a mix between angularjs and reactjs, which the developers are starting to be interested in. And plus the release of the native runtime Capasitor v2, which previously used Cordova. Also as the years have gone on, the devices have become more sophisticated. Whenever I build a plugin or other tool, it's still useful to speed up other projects (frontend, backend, react native, vue native & other nodejs), unlike the standalone flutter with Dart. I bet more on ionic development.

We looked at Capacitor but it is plagued by the same issues Cordova has: it still runs in the WebView and calls the hardware (like the camera) using Intents (t.ly/bAEA). Capacitor uses the same approach Cordova uses, and that crashes randomly on some devices... the issue has been going on for years and there is not any solution with the hybrid approach -> t.ly/AbCv. This is just an example, we have other issues with our app (Ionic + Cordova), we thought about refactoring to react-native but it looks like Flutter is the next big thing by looking at the companies investing in it (Microsoft, Google, Toyota to name a few). Anyway if an app does not use the camera or background services, Ionic + Capacitor is a solid choice.

**Comment by aacassandra** [comment]
> > @mirko77 , Yes, you are right 🎉 . but since Vuejs released version 3, Ionic released a stable version of support for @ ionic/vue. We know that vue is a mix between angularjs and reactjs, which the developers are starting to be interested in. And plus the release of the native runtime Capasitor v2, which previously used Cordova. Also as the years have gone on, the devices have become more sophisticated. Whenever I build a plugin or other tool, it's still useful to speed up other projects (frontend, backend, react native, vue native & other nodejs), unlike the standalone flutter with Dart. I bet more on ionic development.
> 
> We looked at Capacitor but it is plagued by the same issues Cordova has: it still runs in the WebView and calls the hardware (like the camera) [using Intents](t.ly/mtV1) . Capacitor uses the same approach Cordova uses, and that crashes randomly on some devices... the issue has been going on for years and [there is not any solution with the hybrid approach](https://github.com/apache/cordova-plugin-camera/issues/345). This is just an example, we have other issues with our app (Ionic + Cordova), we thought about refactoring to react-native but it looks like Flutter is the next big thing by looking at the companies investing in it (Microsoft, Google, Toyota to name a few). Anyway if an app does not use the camera or background services or the local SQLite database, Ionic + Capacitor is a solid choice.
> 
> Another example would be the date picker [broken by the iOS 14 update](https://stackoverflow.com/questions/64274724/issue-with-html5-date-component-not-working-with-cordova-on-ios-14)... That means we have to do a major update to our app just to fix a date picker... our team is fed up with fixing these things.

I hope you have a good day, but so far I have no problem with Ionic. I have worked for clients, such as integration with the camera plugin, sqlite, background service use [fourground service](https://ionicframework.com/docs/native/foreground-service). the date picker also works perfectly for me both on android and on ios. I only use ion-datetime and momentjs components. My friend is a flutter developer, we also often compare the performance between flutter and ionic, and it's not that much different. I think performance is determined by the coding style of each developer

**Comment by mirko77** [comment]
> > > @mirko77 , Yes, you are right 🎉 . but since Vuejs released version 3, Ionic released a stable version of support for @ ionic/vue. We know that vue is a mix between angularjs and reactjs, which the developers are starting to be interested in. And plus the release of the native runtime Capasitor v2, which previously used Cordova. Also as the years have gone on, the devices have become more sophisticated. Whenever I build a plugin or other tool, it's still useful to speed up other projects (frontend, backend, react native, vue native & other nodejs), unlike the standalone flutter with Dart. I bet more on ionic development.
> > 
> > 
> > We looked at Capacitor but it is plagued by the same issues Cordova has: it still runs in the WebView and calls the hardware (like the camera) [using Intents](t.ly/mtV1) . Capacitor uses the same approach Cordova uses, and that crashes randomly on some devices... the issue has been going on for years and [there is not any solution with the hybrid approach](https://github.com/apache/cordova-plugin-camera/issues/345). This is just an example, we have other issues with our app (Ionic + Cordova), we thought about refactoring to react-native but it looks like Flutter is the next big thing by looking at the companies investing in it (Microsoft, Google, Toyota to name a few). Anyway if an app does not use the camera or background services or the local SQLite database, Ionic + Capacitor is a solid choice.
> > Another example would be the date picker [broken by the iOS 14 update](https://stackoverflow.com/questions/64274724/issue-with-html5-date-component-not-working-with-cordova-on-ios-14)... That means we have to do a major update to our app just to fix a date picker... our team is fed up with fixing these things.
> 
> I hope you have a good day, but so far I have no problem with Ionic. I have worked for clients, such as integration with the camera plugin, sqlite, background service use [fourground service](https://ionicframework.com/docs/native/foreground-service). the date picker also works perfectly for me both on android and on ios or ios 14 tested work 100%. I only use ion-datetime and momentjs components. My friend is a flutter developer, we also often compare the performance between flutter and ionic, and it's not that much different. I think performance is determined by the coding style of each developer

I posted the issues, you might have a read at those. Happy you do not have any of those issues. We do, and our users are not happy.

**Comment by aacassandra** [comment]
![Screen Shot 2021-03-24 at 00 46 49](https://user-images.githubusercontent.com/29236058/112193422-72fb5d80-8c3a-11eb-806f-a58869c5a9ce.png)

**Comment by aacassandra** [comment]
![Screen Shot 2021-03-24 at 00 46 56](https://user-images.githubusercontent.com/29236058/112193511-89091e00-8c3a-11eb-95fa-e60cc5345fe8.png)
